### ISSUE TITLE
MyMarkets does not use replaceName for assets

### ISSUE THREAD

**Issue by sschiessl-bcp** [bitshares/bitshares-ui#1803]
In MyMarkets overview

![image](https://user-images.githubusercontent.com/33128181/44650610-5695d680-a9e7-11e8-8d24-adb509a53307.png)

symbols are directly printed (for base and quote), without using utils.replaceName or AssetName component, see

`https://github.com/bitshares/bitshares-ui/blob/1ee0ba3264abb63a3c06353e35731e80c7aee56d/app/components/Exchange/MyMarkets.jsx#L1048`

**Comment by startailcoon** [comment]
I'm not sure what the problem is, but the value on L1048
https://github.com/bitshares/bitshares-ui/blob/1ee0ba3264abb63a3c06353e35731e80c7aee56d/app/components/Exchange/MyMarkets.jsx#L1048

Comes from 
https://github.com/bitshares/bitshares-ui/blob/1ee0ba3264abb63a3c06353e35731e80c7aee56d/app/components/Exchange/MyMarkets.jsx#L1035

preferedBases comes from Settings and is defined on
https://github.com/bitshares/bitshares-ui/blob/1ee0ba3264abb63a3c06353e35731e80c7aee56d/app/components/Exchange/MyMarkets.jsx#L1192

PreferedBases is just a list of token names stored in settings, like `["BTC","BTS","USD","WLS","OPEN.BTC"]`. These names are then used for listing all available markets for that asset. The BTC tab lists markets like BTC/BTS, BTC/BTDIGE.BCO etc....

**Comment by startailcoon** [comment]
I guess your referring to the header tabs that is printed with L1048, that you refer to, isn't using using the asset replace. Then I understand and agree.

It should be used so assets like open.BTC is viewed correctly.

**Comment by svk31** [comment]
It doesn't make sense to use replaceName or AssetName here, because the tabs actually show all x.BTC assets etc. Even for USD, OPEN.USD is included as well as any future USD assets by one of the supported gateways.

**Comment by sschiessl-bcp** [comment]
Yes, but only because you did a special hack to treat OPEN.BTC as BTC.

Following points:

1. This needs to be refined such that you give it a wildcard symbol (*BTC) as base, which means that all similar assets are shown (bitBTC, OPEN.BTC, GDEX.BTC etc.). The implicit nature of simply putting BTC and it searches for all is confusing, some users might want to actually specify only one of those. This immediately brings up the question how to deal with the UX here. I would say currently no one understands the logic. Example is here:
![image](https://user-images.githubusercontent.com/33128181/44707860-f6b23500-aaa5-11e8-98c8-43c01c5b854f.png)
No one can see why there two BTS rows. If you click on both you will of course see (if you look at the exact market) that it is in fact GDEX.BTC and OPEN.BTC. I do love this feature, but the current UX does not carry it forward

2. The actual definition of which assets are grouped is through available Gateways, which is fine

3. replaceName should be applied due to the possibility to hide namespaces through branding.js

**Comment by svk31** [comment]
1. Not just OPEN.BTC, all the gateways returned by getPossibleGatewayPrefixes here: https://github.com/bitshares/bitshares-ui/blob/develop/app/lib/common/gateways.js#L82

I'm open to suggestions as to how to visually indicate what gateway the market belongs to in the case of "duplicate" markets though, and there's already an issue discussing that.

3. replaceName/AssetName is applied in the rows already.

**Comment by sschiessl-bcp** [comment]
Ah sorry, I mixed it up with the new Dashboard feature. 

We define

https://github.com/bitshares/bitshares-ui/blob/f66bd9675db538ecc645b8e3965a738907634445/app/branding.js#L79-L81

which should correlate to the bitAssets. If you would e.g. replace BTC with OPEN.BTC in branding.js, it would be displayed as OPEN.BTC, even if "OPEN." is a hidden namespace. In my opinion, BTC should lead to only displaying bitBTC. If you want grouped, it should be *BTC in the settings to distinguish the logic (also important if you want to brand your wallet to only show asset from one gateway).

**Comment by sschiessl-bcp** [comment]
> I'm open to suggestions as to how to visually indicate what gateway the market belongs to in the case of "duplicate" markets though, and there's already an issue discussing that.

Could you point me to it please?

**Comment by svk31** [comment]
https://github.com/bitshares/bitshares-ui/issues/1746

**Comment by svk31** [comment]
I think it's very important to group markets this way in order to provide access to all liquid markets, and it also makes it easier for users to actually find the best, most liquid markets.

**Comment by sschiessl-bcp** [comment]
Certainly. I am not saying it should not be on by default. I am saying it should not abuse the bitAsset symbols to do so, but rather be explicitly set.

**Comment by sschiessl-bcp** [comment]
Found a new issue with the implicit approach by looking into ETH. 

There is actually an asset called ETH, which is neither a bitAsset or from a trusted gateway. It is highlighted in the current reference UI
   https://wallet.bitshares.org/#/market/OPEN.DOGE_ETH

Having said that, the logic should be even more refined that you can either
1. Set a wildcard to include all, but the base symbol without prefix must be a bitAsset to be shown
2. Set a specific list of grouped symbols

Examples for:
1. '*BTC' to cover ['BTC', 'GDEX.BTC', ...], but '*ETC' only covers [RUDEX.ETH, GDEX.ETH, ...] since ETH is not a bitAsset
2. Set BTC = ['BTC', 'OPEN.BTC', ...] and ETH = ['RUDEX.ETH', 'GDEX.ETH',  ...]

My vote would be for 1. since I don't see any other way that a non-prefixed asset should be trusted, other than being a bitAsset.

**Comment by happyconcepts** [comment]
> want to brand your wallet to only show asset from one gateway 

Why so much effort toward curating and controlling the content of the blockchain?

This attitude seems precariously exclusionary to me.

**Comment by sschiessl-bcp** [comment]
@happyconcepts 
See #1613

**Comment by happyconcepts** [comment]
> I find the issue with ETH quite crucial actually

Actually, since you are meeting in Amsterdam with George and bitSpark very shortly, and apparently he is launching Spark DEX tomorrow ...

I am the issuer of Spark UIA on BitShares, so the same "crucial" conflict must logically exist for Spark Dex and my coin. 

And I am simply not delighted at being left out of searches or lists of assets since you are suggesting to redesign the BitShares oftware to do so.

**Comment by sschiessl-bcp** [comment]
> And I am simply not delighted at being left out of searches or lists of assets since you are suggesting to redesign the BitShares oftware to do so.

On the dashboard of the Reference UI markets of bitAssets and included Gateways are highlighted. No other private asset is being left out, as no other private asset *should* be included. That the dashboard potetntially shows ETH markets (markets that include the symbol ETH, which is private, no gateway and no bitAsset) is merely a mistake.

**Comment by happyconcepts** [comment]
How is it you choose to speak so, shall we say imperially, when I am trying to have a discussion. It feels very one sided.

**Comment by happyconcepts** [comment]
> No other private asset is being left out, as no other private asset *should* be included. 

This. I believe you need to fundamentally change this attitude, for the best of BitShares. 

It's 6:30am here, please unban me from telegram if you want to discuss it further.

**Comment by happyconcepts** [comment]
> since I don't see any other way that a non-prefixed asset should be trusted, other than being a bitAsset.

You will never trust my Spark coin, then, because it is a non prefixed UIA? 

Or just because I came here as Spark to Bitshares before bitSpark?

**Comment by sschiessl-bcp** [comment]
> Actually, since you are meeting in Amsterdam with George and bitSpark very shortly, and apparently he is launching Spark DEX tomorrow 30 Aug.

Will you be in Amsterdam too @happyconcepts ?

> imperially

I don't even know how to do that

> This. I believe you need to fundamentally change this attitude, for the best of BitShares.

I am only talking about highlighting it on the Dashboard. The reference UI allows to search and display any asset. I suppose the UI could do some promotion for assets on the Dashboard, but that would need to be a completely new format.

> It's 6:30am here, please unban me from telegram if you want to discuss it further.

What is your telegram handle and which channel are you talking?

> You will never trust my Spark coin, then, because it is a non prefixed UIA?

The reference UI highlights commonly known crypto-currencies (BTC, ETH, ...) on the Dashboard. Trusted was maybe the wrong wording. Let's say "not promoting".

> Or just because I came here as Spark to Bitshares before bitSpark?

That is a mere name collission. bitSpark is also not highlighted.

I don't see that this issue touches your SPARK asset at all. What is your suggestion now to do differently=

**Comment by happyconcepts** [comment]
> I don't see that this issue touches your SPARK asset at all. What is your suggestion now to do differently=

Well there is a private offering so I will be here on BitShares with Spark for at least a little longer now.

I must have somehow missed the invitation to Amsterdam. I am Spark token on telegram, banned from the BitShares group last summer within just two weeks of joining. 

Maybe to make way for bitspark's arrival, or maybe coincidence, or maybe unfairly branded as an enemy-of-the-state, who knows? But there were also two attempts on my life that week, and my founding partner for Spark was shot dead Aug 5 so I had larger issues to attend to than to fight the paranoid-ban at the time. The rest is just mob mentality.

**Comment by wmbutler** [comment]
@sschiessl-bcp I support where you are going with this. It's important that users be able to easily identify trusted gateways and their associated tokens that represent real world crypto. @happyconcepts you are also able to reserve a prefix and do the same.

**Comment by clockworkgr** [comment]
@sschiessl-bcp I too like the wildcard approach.

Is this being worked on?

**Comment by sschiessl-bcp** [comment]
Not by me atm
